api/envoy/config/route/v3/route_components.proto

```diff
@@ -542,6 +542,19 @@ message RouteMatch {
     //
     // Expect the value to not contain ``?`` or ``#`` and not to end in ``/``
     string path_separated_prefix = 14 [(validate.rules).string = {pattern: "^[^?#]+[^?#/]$"}];
+
+    // If specified, the route is a template match rule meaning that the
+    // ``:path`` header (without the query string) must match the given
+    // ``path_template_match`` pattern.
+    //
+    // For example:
```

Comment:
These examples are good, but I think it would be worth explicitly stating what * means, what ** means, and what {foo} is, instead of only relying on the examples.